### ISSUE TITLE
Pipelines in conditionals not working

### ISSUE THREAD

**Issue by glyh** [oilshell/oil#1876]
```ysh
if ls -la | head {
>  echo yes
> }
  if ls -la | head {
            ^
[ interactive ]:1: fatal: strict_errexit only allows simple commands in conditionals (got command.Pipeline). 
```

**Comment by andychu** [comment]
This is intended -- the idiom is to use `try`

which is somewhere in here

https://www.oilshell.org/release/0.21.0/doc/error-handling.html

or here

https://www.oilshell.org/release/latest/doc/idioms.html

I think we need to hook it up to the new **error catalog**, mentioned in the latest release

i.e. add an error code and explanatory message

**Comment by Subhashis-1** [comment]
Hi, I’d like to work on this issue. I’m new to the codebase — please let me know if this sounds good.

**Comment by andychu** [comment]
Yes this is a documentation issue - it can be a good place to get started!  

Most discussion happens on https://oilshell.zulipchat.com/

I might look at some example commits in the error catalog, e.g. `git log doc/error-catalog.md`

**Comment by Subhashis-1** [comment]
> Yes this is a documentation issue - it can be a good place to get started!
> 
> Most discussion happens on https://oilshell.zulipchat.com/
> 
> I might look at some example commits in the error catalog, e.g. `git log doc/error-catalog.md`

Hi @andychu, thanks! I’ve added a new error entry, OILS-ERR-303, to the error catalog documenting strict_errexit for compound commands in if statements. 
The PR is ready for review: https://github.com/Subhashis-1/oils/pull/new/add-oils-err-303
